Bizarre malloc error in `CalculateCurvature`

I've finally gotten to pulling down the latest master and started updating the C bindings and have hit a bit of a snag. Everything compiles and tests fine for me with CUDA enabled, leading to [`mallocManaged`](https://github.com/elalish/manifold/blob/cce35aa5a04ec606f00dad6d82d66453148128d9/src/utilities/include/vec_dh.h#L254) using `cudaMallocManaged` instead of `malloc`. So malloc is being tripped up by size parameters that are doing fine in the CUDA branch. Any idea what could be going wrong? Won't have more time to try and figure it out for a bit, so dumping here to see if it makes sense to you guys.

**Build Error:**
```
In file included from /home/geoff/git/manifold/src/utilities/include/sparse.h:22,
                 from /home/geoff/git/manifold/src/collider/include/collider.h:17,
                 from /home/geoff/git/manifold/src/manifold/src/impl.h:18,
                 from /home/geoff/git/manifold/src/manifold/src/properties.cpp:21:
In static member function ‘static void manifold::ManagedVec<T>::mallocManaged(T**, size_t) [with T = glm::vec<3, int, glm::packed_highp>]’,
    inlined from ‘void manifold::ManagedVec<T>::shrink_to_fit() [with T = glm::vec<3, int, glm::packed_highp>]’ at /home/geoff/git/manifold/src/utilities/include/vec_dh.h:169:20,
    inlined from ‘void manifold::VecDH<T>::resize(int, T) [with T = glm::vec<3, int, glm::packed_highp>]’ at /home/geoff/git/manifold/src/utilities/include/vec_dh.h:349:36,
    inlined from ‘void manifold::Manifold::Impl::CalculateCurvature(int, int)’ at /home/geoff/git/manifold/src/manifold/src/properties.cpp:355:39:
/home/geoff/git/manifold/src/utilities/include/vec_dh.h:260:42: error: argument 1 range [18446744065119617032, 18446744073709551604] exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
  260 |       *ptr = reinterpret_cast<T *>(malloc(bytes));
      |                                    ~~~~~~^~~~~~~
In file included from /usr/include/c++/13.1.1/cstdlib:79,
                 from /usr/include/c++/13.1.1/ext/string_conversions.h:43,
                 from /usr/include/c++/13.1.1/bits/basic_string.h:4110,
                 from /usr/include/c++/13.1.1/string:54,
                 from /usr/include/c++/13.1.1/bits/locale_classes.h:40,
                 from /usr/include/c++/13.1.1/bits/ios_base.h:41,
                 from /usr/include/c++/13.1.1/streambuf:43,
                 from /usr/include/c++/13.1.1/bits/streambuf_iterator.h:35,
                 from /usr/include/c++/13.1.1/iterator:66,
                 from /home/geoff/git/manifold/src/third_party/thrust/thrust/iterator/iterator_traits.h:36,
                 from /home/geoff/git/manifold/src/third_party/thrust/thrust/count.h:26,
                 from /home/geoff/git/manifold/src/manifold/src/properties.cpp:15:
/usr/include/stdlib.h: In member function ‘void manifold::Manifold::Impl::CalculateCurvature(int, int)’:
/usr/include/stdlib.h:553:14: note: in a call to allocation function ‘void* malloc(size_t)’ declared here
  553 | extern void *malloc (size_t __size) __THROW __attribute_malloc__
      |              ^~~~~~
```

**Machine:**
```
OS: EndeavourOS Linux (Arch based)
Kernel: 6.3.7-zen1-1-ze
```

--- a/src/utilities/include/vec_dh.h
+++ b/src/utilities/include/vec_dh.h
@@ -166,8 +166,9 @@ class ManagedVec {
   void shrink_to_fit() {
     T *newBuffer = nullptr;
     if (size_ > 0) {
-      mallocManaged(&newBuffer, size_ * sizeof(T));
-      prefetch(newBuffer, size_ * sizeof(T), onHost);
+      int n_bytes = size_ * sizeof(T);
+      mallocManaged(&newBuffer, n_bytes);
+      prefetch(newBuffer, n_bytes, onHost);
       uninitialized_copy(autoPolicy(size_), ptr_, ptr_ + size_, newBuffer);
     }
     freeManaged(ptr_);

